WhitespaceAfter does not work on 'switch' and lambda statements

I want to be able to configure Checkstyle to report a violation on missing space after 'switch', which is currently not possible.
WhitespaceAfter detects a missing space after 'for', 'if' etc. but _not_ after 'switch'.
It would be nice if there would be a LITERAL_SWITCH similar to LITERAL_FOR etc.

https://checkstyle.org/config_whitespace.html#WhitespaceAfter
I have downloaded the latest cli from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

**How it works Now:**

```bash
peter@pangolin:~/Downloads/checkstyle$ javac Test.java 

peter@pangolin:~/Downloads/checkstyle$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
	"-//Puppy Crawl//DTD Check Configuration 1.3//EN"
	"http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd">
<module name="Checker">
	<module name="TreeWalker">
		<module name="WhitespaceAfter"/>
	</module>
</module>

peter@pangolin:~/Downloads/checkstyle$ cat Test.java 
class Test {

	void test() {
		for(int i = 0; i < 10; i++) { // Checkstyle catches this, configurable with LITERAL_FOR.
		}
		switch(1) { // Checkstyle does NOT catch this, there is no LITERAL_SWITCH to configure.
			default:
				break;
		}
	}
}

peter@pangolin:~/Downloads/checkstyle$ RUN_LOCALE="-Duser.language=en -Duser.country=US"
peter@pangolin:~/Downloads/checkstyle$ java $RUN_LOCALE -jar checkstyle-9.3-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/peter/Downloads/checkstyle/Test.java:4:17: 'for' is not followed by whitespace. [WhitespaceAfter]
Audit done.
Checkstyle ends with 1 errors.
```

The problem is that WhitespaceAfter catches missing spaces after 'if', 'for' etc., but _not_ after 'switch'.
Similar to 'for', I suggest a configurable token LITERAL_SWITCH with the "token" property which is configured by default, and a similar output:

```bash
[ERROR] home/peter/Downloads/checkstyle/Test.java:6:21: 'switch' is not followed by whitespace. [WhitespaceAfter]
```

**Additional context**

The code style that I want to impose:

```java
for (int i = 0; i < 10; i++) {
}
switch (1) {
	default:
		break;
}
```

--- a/.ci/validation.sh
+++ b/.ci/validation.sh
@@ -608,7 +608,7 @@ no-error-orekit)
   # no CI is enforced in project, so to make our build stable we should
   # checkout to latest release/development (annotated tag or hash) or sha that have fix we need
   # git checkout $(git describe --abbrev=0 --tags)
-  git checkout "76760bf""bf""b847e227490cd5d3662f""ca087f1a324"
+  git checkout "851de782c6""d16d""f725fa""bb4646ff0dd086723415"
   mvn -e --no-transfer-progress compile checkstyle:check \
     -Dorekit.checkstyle.version=${CS_POM_VERSION}
   cd ..

--- a/config/suppressions.xml
+++ b/config/suppressions.xml
@@ -6,7 +6,7 @@
 
 <suppressions>
   <!-- can't split long messages between lines -->
-  <suppress id="lineLengthXml" files="google_checks\.xml" lines="56,126"/>
+  <suppress id="lineLengthXml" files="google_checks\.xml" lines="56,127"/>
   <!-- don't validate generated files -->
   <suppress id="lineLengthXml" files="releasenotes_old_6\-0_7\-8\.xml"/>
   <suppress id="lineLengthXml" files="releasenotes_old_1\-0_5\-9\.xml"/>

--- a/src/it/java/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/WhitespaceAfterTest.java
+++ b/src/it/java/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/WhitespaceAfterTest.java
@@ -53,6 +53,9 @@ public class WhitespaceAfterTest extends AbstractGoogleModuleTestSupport {
             "14:9: " + getCheckMessage(clazz, message, "if"),
             "14:18: " + getCheckMessage(clazz, message, "typecast"),
             "17:9: " + getCheckMessage(clazz, message, "else"),
+            "22:28: " + getCheckMessage(clazz, message, "..."),
+            "23:26: " + getCheckMessage(clazz, message, "->"),
+            "24:9: " + getCheckMessage(clazz, message, "switch"),
         };
         final Configuration checkConfig = getModuleConfig("WhitespaceAfter");
         final String filePath = getPath("InputWhitespaceAfterBad.java");

--- a/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterBad.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterBad.java
@@ -18,4 +18,12 @@ public class InputWhitespaceAfterBad {
             System.out.println("false");
         }
     }
+
+    public void check3 (int...a) { // warn
+        Runnable r2 = () ->String.valueOf("Hello world two!"); // warn
+        switch(a[0]) { // warn
+            default:
+                break;
+        }
+    }
 }

--- a/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterGood.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterGood.java
@@ -18,4 +18,12 @@ public class InputWhitespaceAfterGood {
             System.out.println("false");
         }
     }
+
+    public void check3(int... a) {
+        Runnable r2 = () -> String.valueOf("Hello world two!");
+        switch (a[0]) {
+            default:
+                break;
+        }
+    }
 }

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java
@@ -57,7 +57,11 @@ import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
  * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_DO">
  * DO_WHILE</a>,
  * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ELLIPSIS">
- * ELLIPSIS</a>.
+ * ELLIPSIS</a>,
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_SWITCH">
+ * LITERAL_SWITCH</a>,
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LAMBDA">
+ * LAMBDA</a>.
  * </li>
  * </ul>
  * <p>
@@ -153,6 +157,8 @@ public class WhitespaceAfterCheck
             TokenTypes.LITERAL_FOR,
             TokenTypes.DO_WHILE,
             TokenTypes.ELLIPSIS,
+            TokenTypes.LITERAL_SWITCH,
+            TokenTypes.LAMBDA,
         };
     }
 

--- a/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/whitespace/WhitespaceAfterCheck.xml
+++ b/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/whitespace/WhitespaceAfterCheck.xml
@@ -11,7 +11,7 @@
  EmptyForIteratorPad&lt;/a&gt; to validate empty for iterators.
  &lt;/p&gt;</description>
          <properties>
-            <property default-value="COMMA,SEMI,TYPECAST,LITERAL_IF,LITERAL_ELSE,LITERAL_WHILE,LITERAL_DO,LITERAL_FOR,DO_WHILE,ELLIPSIS"
+            <property default-value="COMMA,SEMI,TYPECAST,LITERAL_IF,LITERAL_ELSE,LITERAL_WHILE,LITERAL_DO,LITERAL_FOR,DO_WHILE,ELLIPSIS,LITERAL_SWITCH,LAMBDA"
                       name="tokens"
                       type="java.lang.String[]"
                       validation-type="tokenSet">

--- a/src/main/resources/google_checks.xml
+++ b/src/main/resources/google_checks.xml
@@ -105,7 +105,8 @@
     <module name="WhitespaceAfter">
       <property name="tokens"
                value="COMMA, SEMI, TYPECAST, LITERAL_IF, LITERAL_ELSE,
-                    LITERAL_WHILE, LITERAL_DO, LITERAL_FOR, DO_WHILE, ELLIPSIS"/>
+                    LITERAL_WHILE, LITERAL_DO, LITERAL_FOR, DO_WHILE, ELLIPSIS,
+                    LITERAL_SWITCH, LAMBDA"/>
     </module>
     <module name="WhitespaceAround">
       <property name="allowEmptyConstructors" value="true"/>

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheckTest.java
@@ -198,4 +198,33 @@ public class WhitespaceAfterCheckTest
         };
         verifyWithInlineConfigParser(getPath("InputWhitespaceAfterVarargs.java"), expected);
     }
+
+    @Test
+    public void testSwitchStatements() throws Exception {
+        final String[] expected = {
+            "18:9: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "switch"),
+            "30:9: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "switch"),
+            "32:21: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "->"),
+            "39:9: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "switch"),
+            "40:27: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "->"),
+            "41:28: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "->"),
+            "48:9: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "switch"),
+        };
+
+        verifyWithInlineConfigParser(
+                getNonCompilablePath("InputWhitespaceAfterSwitchStatements.java"),
+                expected);
+    }
+
+    @Test
+    public void testLambdaExpressions() throws Exception {
+        final String[] expected = {
+            "17:29: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "->"),
+            "19:29: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "->"),
+            "27:49: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "->"),
+        };
+
+        verifyWithInlineConfigParser(getPath("InputWhitespaceAfterLambdaExpressions.java"),
+                expected);
+    }
 }

--- a/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterSwitchStatements.java
+++ b/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterSwitchStatements.java
@@ -1,0 +1,58 @@
+/*
+WhitespaceAfter
+tokens = LITERAL_SWITCH, LAMBDA
+
+*/
+
+//non-compiled with javac: Compilable with Java17
+package com.puppycrawl.tools.checkstyle.checks.whitespace.whitespaceafter;
+
+public class InputWhitespaceAfterSwitchStatements {
+    enum Day {
+        NONE, SAT, SUN, MON,
+        TUS, WED, THU, FRI
+    }
+
+    String whatIsToday(Day day) {
+        String today;
+        switch(day) { // violation
+            case SAT, SUN:
+                today = "Weekend day";
+                break;
+            default:
+                throw new IllegalArgumentException("Invalid day: " + day.name());
+        }
+        return today;
+    }
+
+    String whatIsToday2(Day day) {
+        var today = "";
+        switch(day) { // violation
+            case MON, TUS, WED, THU, FRI -> today = "Working day";
+            default ->throw new IllegalArgumentException(); // violation
+        }
+        return today;
+    }
+
+    void switchWithPatternMatching() {
+        Object o = "";
+        switch(o) { // violation
+            case String s ->s.trim(); // violation
+            case Integer a ->a.intValue(); // violation
+            default -> throw new IllegalArgumentException(); // ok
+        }
+    }
+
+    void normalSwitch() {
+        final int a = 12;
+        switch(a) { // violation
+            case 1:
+                break;
+            case 2:
+            case 3:
+                break;
+            default:
+                break;
+        }
+    }
+}

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterLambdaExpressions.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterLambdaExpressions.java
@@ -1,0 +1,29 @@
+/*
+WhitespaceAfter
+tokens = LAMBDA
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.whitespace.whitespaceafter;
+
+import java.util.function.Function;
+
+public class InputWhitespaceAfterLambdaExpressions {
+
+    static Runnable r1 = () -> { // ok
+        String.valueOf("Hello world one!");
+    };
+
+    static Runnable r2 = () ->String.valueOf("Hello world two!"); // violation
+
+    static Runnable r3 = () ->{String.valueOf("Hello world two!");}; // violation
+
+    static Runnable r4 = () -> // ok
+    {
+        String.valueOf("Hello world one!");
+    };
+
+    public void foo() {
+        Function<Object, String> function = (o) ->o.toString(); // violation
+    }
+}

--- a/src/xdocs/config_whitespace.xml
+++ b/src/xdocs/config_whitespace.xml
@@ -2904,6 +2904,10 @@ class Bar {
                   DO_WHILE</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ELLIPSIS">
                   ELLIPSIS</a>
+                , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_SWITCH">
+                  LITERAL_SWITCH</a>
+                , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LAMBDA">
+                  LAMBDA</a>
                   .
               </td>
 
@@ -2928,6 +2932,10 @@ class Bar {
                   DO_WHILE</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ELLIPSIS">
                   ELLIPSIS</a>
+                , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_SWITCH">
+                  LITERAL_SWITCH</a>
+                , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LAMBDA">
+                  LAMBDA</a>
                   .
               </td>
               <td>3.0</td>

